airgun/widgets.py

```diff
@@ -1322,9 +1406,40 @@ def read(self):
         if not self.has_rows:
             self.logger.debug('Table {} is empty'.format(self.locator))
             return []
-
+        if self.pagination.is_displayed:
+            return self._read_all()
         return super(SatTable, self).read()
 
+    def _read_all(self):
+        """Return all available table values with using pagination navigation."""
+        table_rows = []
+        page_number = 1
+        if self.pagination.current_page != page_number:
+            # goto first page
+            self.pagination.first_page()
+            wait_for(
+                lambda: self.pagination.current_page == page_number,
+                timeout=30,
+                delay=1,
+                logger=self.logger
+            )
+        while page_number <= self.pagination.total_pages:
```

Comment:
better change on `for`, but probably it is not so important

Reply:
have tried with that , it will lead to a more complicated code, the one here is more readable.